File: src/main/java/swf/army/mil/blog/comment/CommentService.java
package swf.army.mil.blog.comment;

import jakarta.persistence.EntityNotFoundException;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CommentService {

    private final CommentRepository commentRepository;

    public CommentService(CommentRepository r) { this.commentRepository = r; }

    public Comment save(Comment c) {
        return commentRepository.save(c);
    }

    public Comment findByID(Long id) {
        return commentRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Comment not found"));
    }

    public List<Comment> findByPostID(Long postID) {
        return commentRepository.findByPostId(postID);
    }

    public void delete(Long id) {
        Comment c = findByID(id);

        commentRepository.delete(c);
    }
}

File: src/main/kotlin/com/personal/blog/comment/CommentService.kt
package com.personal.blog.comment

import com.personal.blog.post.PostService
import org.springframework.stereotype.Service

@Service
class CommentService(val repo: CommentRepository, val postService: PostService) {

    fun create(comment: CreateCommentRequest): Comment {
        val originalPost = postService.retrieve(comment.parent)

        return repo.save(Comment(
            parentPost = originalPost,
            text = comment.text))
    }

    fun retrieveByID(id: Long): Comment {
        val myComment = repo.findById(id)
            .orElseThrow { NoSuchElementException("Comment not found") }

        return myComment
    }

    fun retrieveByPostID(id: Long): List<Comment> = repo.findAllByParentPost_Id(id)

    fun delete(id: Long) {
        val existingComment = repo.findById(id)
            .orElseThrow { NoSuchElementException("Comment not found") }

        repo.delete(existingComment)
    }

    fun deleteByPostID(id: Long) = repo.deleteAllByParentPost_Id(id)
}